hr_expense_invoice/models/hr_expense_sheet.py

```diff
@@ -17,19 +17,12 @@ def get_expense_sheets_with_invoices(self, func):
             lambda sheet: func(expense.invoice_id for expense in sheet.expense_line_ids)
         )
 
-    def _do_create_moves(self):
-        """Don't let super to create any move:
-        - Paid by company: there's already the invoice.
-        - Paid by employee: we create here a journal entry transferring the AP
-          balance from the invoice partner to the employee.
-        """
-        expense_sheets_with_invoices = self.get_expense_sheets_with_invoices(all)
-        res = super(
-            HrExpenseSheet, self - expense_sheets_with_invoices
-        )._do_create_moves()
+    def _do_create_ap_moves(self):
         # Create AP transfer entry for expenses paid by employees
-        for expense in expense_sheets_with_invoices.expense_line_ids:
-            if expense.payment_mode == "own_account":
+        for expense in self.expense_line_ids.filtered(
+            lambda expense: expense.invoice_id
+        ):
```

Comment:
Use the available syntactic sugar:

```suggestion
        for expense in self.expense_line_ids.filtered("invoice_id"):
```